### ISSUE TITLE
Do we have too many YAML libraries?

### ISSUE THREAD

**Issue by john-science** [terrapower/armi#575]
I noticed that in our dependencies we have three different YAML libraries:

* `PyYAML`
* `ruamel`
* `yamlize`

https://github.com/terrapower/armi/blob/db90bc8e27398f840c9e345f0b0e8dd5c7eb69c4/setup.py#L69-L74

Surely we could reduce the number of these, right?

I should note that `ruamel` is a dependency of `yamlize`, so that's where we get that.  In that case, it would seem that `PyYAML` is the odd-one-out, and the easiest to ditch to reduce our dependencies. 

> Can we replace PyYAML with `ruamel.yaml`? Is that convenient?

**Comment by john-science** [comment]
Yes, we can and should remove the `PyYAML` dependency from ARMI.  We are only using it in 3 places (in 2 files) to quickly load up a YAML file into a dictionary:

https://github.com/terrapower/armi/blob/8f52db65ef29d34a7432f339b6a69d05a282d330/armi/nucDirectory/nuclideBases.py#L310-L311

https://github.com/terrapower/armi/blob/8f52db65ef29d34a7432f339b6a69d05a282d330/armi/nucDirectory/nuclideBases.py#L402

https://github.com/terrapower/armi/blob/8f52db65ef29d34a7432f339b6a69d05a282d330/armi/cases/inputModifiers/tests/test_inputModifiers.py#L204

There is nothing complicated, special, or unique about our usage of `PyYAML` that we could 100% drop-in replace with `ruamel.yaml`, which we need anyway as part of `Yamlize`.

> TLDR; we can _easily_ remove one of our three YAML dependencies: PyYAML. And I will l do that.
